pkg/resources/stream.go

```diff
@@ -0,0 +1,283 @@
+		Required:    true,
+		ForceNew:    true,
+		Description: "The database in which to create the stream.",
+	},
+	"comment": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Specifies a comment for the stream.",
+	},
+	"on_table": {
+		Type:        schema.TypeString,
+		Optional:    true,
+		Description: "Name of the table the stream will monitor.",
+	},
+	"append_only": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Default:     false,
+		Description: "Type of the stream that will be created.",
+	},
+	"owner": {
+		Type:        schema.TypeString,
+		Computed:    true,
+		Description: "Name of the role that owns the stream.",
+	},
+}
+
+func Stream() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateStream,
+		Read:   ReadStream,
+		Update: UpdateStream,
+		Delete: DeleteStream,
+
+		Schema: streamSchema,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+type streamID struct {
+	DatabaseName string
+	SchemaName   string
+	StreamName   string
+}
+
+type streamOnTableID struct {
+	DatabaseName string
+	SchemaName   string
+	OnTableName  string
+}
+
+//String() takes in a streamID object and returns a pipe-delimited string:
+//DatabaseName|SchemaName|StreamName
+func (si *streamID) String() (string, error) {
+	var buf bytes.Buffer
+	csvWriter := csv.NewWriter(&buf)
+	csvWriter.Comma = streamIDDelimiter
+	dataIdentifiers := [][]string{{si.DatabaseName, si.SchemaName, si.StreamName}}
+	err := csvWriter.WriteAll(dataIdentifiers)
+	if err != nil {
+		return "", err
+	}
+	strStreamID := strings.TrimSpace(buf.String())
+	return strStreamID, nil
+}
+
+// streamIDFromString() takes in a pipe-delimited string: DatabaseName|SchemaName|StreamName
+// and returns a streamID object
+func streamIDFromString(stringID string) (*streamID, error) {
+	reader := csv.NewReader(strings.NewReader(stringID))
+	reader.Comma = streamIDDelimiter
+	lines, err := reader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("Not CSV compatible")
+	}
+
+	if len(lines) != 1 {
+		return nil, fmt.Errorf("1 line at a time")
+	}
+	if len(lines[0]) != 3 {
+		return nil, fmt.Errorf("3 fields allowed")
+	}
+
+	streamResult := &streamID{
+		DatabaseName: lines[0][0],
+		SchemaName:   lines[0][1],
+		StreamName:   lines[0][2],
+	}
+	return streamResult, nil
+}
+
+// streamOnTableIDFromString() takes in a dot-delimited string: DatabaseName.SchemaName.TableName
+// and returns a streamOnTableID object
+func streamOnTableIDFromString(stringID string) (*streamOnTableID, error) {
+	reader := csv.NewReader(strings.NewReader(stringID))
+	reader.Comma = streamOnTableIDDelimiter
+	lines, err := reader.ReadAll()
+	if err != nil {
+		return nil, fmt.Errorf("Not CSV compatible")
+	}
+
+	if len(lines) != 1 {
+		return nil, fmt.Errorf("1 line at a time")
+	}
+	if len(lines[0]) != 3 {
+		return nil, fmt.Errorf("on table format: database_name.schema_name.target_table_name")
```

Comment:
Could we print out the value of `lines` here to tell the coder what they got wrong? Instead of just stating what it should be?

Reply:
sure, I've updated the comment let me know if you prefer something different.